### DISCUSSION TITLE
Can nginx do transparent compression of websocket connections

### DISCUSSION THREAD

**Discussion by alex-semenyuk** [nginx-proxy/nginx-proxy#2161] [Q&A] [answered]
I have nginx acting as reverse-proxy for an ASP.net

I'd like nginx to do gzip compression (ie, permessage-deflate / rfc7692) for my Websocket connections, but I can't find any config options for that.

Can nginx do that? Are there any plugins to make it work? If no, is there something else I can use?

**Comment by denisandreenko** [comment, marked as answer]
just edit your nginx config. It is the same for static sites, proxies or websockets

```gunzip on; // be sure to include this. This enables runtime decompression for clients that do not accept gzip
gzip on; //enables gzip for request
gzip_proxied any; //enables compression on proxies
gzip_types *; //compress everything there is
```

if this does not work, try compressing the messages befor sending them through the socket. 
You can read more about it here: https://docs.nginx.com/nginx/admin-guide/web-server/compression/
